### PR TITLE
Refactored __hf_index_id and __hf_fts_score

### PR DESCRIPTION
**Pull request by devesh-2002**
Issue #2798,
I have Refactored the __hf_index_id and __hf_fts_score, and added it to constants.py.
@severo, Please look into it and let me know, if there are any changes needed.

### CODE REVIEW

File: services/search/src/search/routes/search.py [hunk 1]
```diff
@@ -49,9 +48,9 @@
 
 logger = logging.getLogger(__name__)
 
-FTS_STAGE_TABLE_COMMAND = "SELECT * FROM (SELECT __hf_index_id, fts_main_data.match_bm25(__hf_index_id, ?) AS __hf_fts_score FROM data) A WHERE __hf_fts_score IS NOT NULL;"
+FTS_STAGE_TABLE_COMMAND = "SELECT * FROM (SELECT {ROW_IDX_COLUMN}, fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) AS {HF_FTS_SCORE} FROM data) A WHERE {HF_FTS_SCORE} IS NOT NULL;"
```
Comment (severo) [inline services/search/src/search/routes/search.py:51]:
```suggestion
FTS_STAGE_TABLE_COMMAND = f"SELECT * FROM (SELECT {ROW_IDX_COLUMN}, fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) AS {HF_FTS_SCORE} FROM data) A WHERE {HF_FTS_SCORE} IS NOT NULL;"
```

File: services/search/src/search/routes/search.py [hunk 2]
```diff
@@ -49,9 +48,9 @@
...
 JOIN_STAGE_AND_DATA_COMMAND = (
-    "SELECT data.* FROM fts_stage_table JOIN data USING(__hf_index_id) ORDER BY fts_stage_table.__hf_fts_score DESC;"
+    "SELECT data.* FROM fts_stage_table JOIN data USING({ROW_IDX_COLUMN}) ORDER BY fts_stage_table.{HF_FTS_SCORE} DESC;"
```
Comment (severo) [inline services/search/src/search/routes/search.py:53]:
```suggestion
    f"SELECT data.* FROM fts_stage_table JOIN data USING({ROW_IDX_COLUMN}) ORDER BY fts_stage_table.{HF_FTS_SCORE} DESC;"
```

File: services/search/tests/routes/test_filter.py [hunk 3]
```diff
@@ -12,12 +12,12 @@
...
 from datasets import Dataset
 from libapi.exceptions import InvalidParameterError
-from libapi.response import ROW_IDX_COLUMN, create_response
+from libapi.response import create_response
 from libcommon.storage_client import StorageClient
 
 from search.config import AppConfig
 from search.routes.filter import execute_filter_query, validate_query_parameter
-
+from libcommon.constants import ROW_IDX_COLUMN
```
Comment (severo) [inline services/search/tests/routes/test_filter.py:20]: Can you run `make install; make style` in `services/search`, to fix the linting issue (we automatically order the imports)? I can do it for you if you prefer.

File: services/search/tests/routes/test_filter.py [hunk 4]
```diff
@@ -53,10 +53,10 @@ def index_file_location(ds: Dataset) -> Generator[str, None, None]:
     con.execute("LOAD 'fts';")
     con.sql("CREATE OR REPLACE SEQUENCE serial START 0 MINVALUE 0;")
     sample_df = ds.to_pandas()  # noqa: F841
-    create_command_sql = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS __hf_index_id, * FROM sample_df"
+    create_command_sql = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS {ROW_IDX_COLUMN}, * FROM sample_df"
```
Comment (severo) [inline services/search/tests/routes/test_filter.py:56]:
```suggestion
    create_command_sql = f"CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS {ROW_IDX_COLUMN}, * FROM sample_df"
```

File: services/search/tests/routes/test_filter.py [hunk 5]
```diff
@@ -53,10 +53,10 @@ def index_file_location(ds: Dataset) -> Generator[str, None, None]:
...
     con.sql(create_command_sql)
     # assert sample_df.shape[0] == con.execute(query="SELECT COUNT(*) FROM data;").fetchall()[0][0]
-    con.sql("PRAGMA create_fts_index('data', '__hf_index_id', 'name', 'gender', overwrite=1);")
+    con.sql("PRAGMA create_fts_index('data', ROW_IDX_COLUMN, 'name', 'gender', overwrite=1);")
```
Comment (severo) [inline services/search/tests/routes/test_filter.py:59]:
```suggestion
    con.sql(f"PRAGMA create_fts_index('data', ROW_IDX_COLUMN, 'name', 'gender', overwrite=1);")
```

File: services/search/tests/routes/test_search.py [hunk 6]
```diff
@@ -12,7 +12,7 @@
 from libcommon.storage import StrPath
 
 from search.routes.search import full_text_search
-
+from libcommon.constants import ROW_IDX_COLUMN
```
Comment (severo) [inline services/search/tests/routes/test_search.py:15]: same

File: services/worker/src/worker/job_runners/split/duckdb_index.py [hunk 7]
```diff
@@ -50,10 +51,10 @@
 DATASET_TYPE = "dataset"
 DUCKDB_DEFAULT_INDEX_FILENAME = "index.duckdb"
 DUCKDB_DEFAULT_PARTIAL_INDEX_FILENAME = "partial-index.duckdb"
-CREATE_INDEX_COMMAND = "PRAGMA create_fts_index('data', '__hf_index_id', {columns}, overwrite=1);"
+CREATE_INDEX_COMMAND = "PRAGMA create_fts_index('data', {ROW_IDX_COLUMN}, {columns}, overwrite=1);"
```
Comment (severo) [inline services/worker/src/worker/job_runners/split/duckdb_index.py:54]:
```suggestion
CREATE_INDEX_COMMAND = f"PRAGMA create_fts_index('data', {ROW_IDX_COLUMN}, {columns}, overwrite=1);"
```

File: services/worker/src/worker/job_runners/split/duckdb_index.py [hunk 8]
```diff
@@ -50,10 +51,10 @@
...
 CREATE_TABLE_COMMAND = "CREATE OR REPLACE TABLE data AS SELECT {columns} FROM '{source}';"
 CREATE_SEQUENCE_COMMAND = "CREATE OR REPLACE SEQUENCE serial START 0 MINVALUE 0;"
-ALTER_TABLE_BY_ADDING_SEQUENCE_COLUMN = "ALTER TABLE data ADD COLUMN __hf_index_id BIGINT DEFAULT nextval('serial');"
+ALTER_TABLE_BY_ADDING_SEQUENCE_COLUMN = "ALTER TABLE data ADD COLUMN {ROW_IDX_COLUMN} BIGINT DEFAULT nextval('serial');"
```
Comment (severo) [inline services/worker/src/worker/job_runners/split/duckdb_index.py:57]:
```suggestion
ALTER_TABLE_BY_ADDING_SEQUENCE_COLUMN = f"ALTER TABLE data ADD COLUMN {ROW_IDX_COLUMN} BIGINT DEFAULT nextval('serial');"
```

File: services/worker/tests/job_runners/split/test_duckdb_index.py [hunk 9]
```diff
@@ -39,6 +39,7 @@
 
 from ...fixtures.hub import HubDatasetTest
 from ..utils import REVISION_NAME
+from libcommon.constants import ROW_IDX_COLUMN, HF_FTS_SCORE
```
Comment (severo) [inline services/worker/tests/job_runners/split/test_duckdb_index.py:42]: same

File: services/worker/tests/job_runners/split/test_duckdb_index.py [hunk 10]
```diff
@@ -378,7 +379,7 @@ def test_compute(
             # perform a search to validate fts feature
             query = "Lord Vader"
             result = con.execute(
-                "SELECT __hf_index_id, text FROM data WHERE fts_main_data.match_bm25(__hf_index_id, ?) IS NOT NULL;",
+                "SELECT {ROW_IDX_COLUMN}, text FROM data WHERE fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) IS NOT NULL;",
```
Comment (severo) [inline services/worker/tests/job_runners/split/test_duckdb_index.py:382]:
```suggestion
                f"SELECT {ROW_IDX_COLUMN}, text FROM data WHERE fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) IS NOT NULL;",
```

File: services/worker/tests/job_runners/split/test_duckdb_index.py [hunk 11]
```diff
@@ -479,8 +480,8 @@ def test_get_indexable_columns(features: Features, expected: list[str]) -> None:
 
 
 FTS_COMMAND = (
-    "SELECT * EXCLUDE (__hf_fts_score) FROM (SELECT *, fts_main_data.match_bm25(__hf_index_id, ?) AS __hf_fts_score"
-    " FROM data) A WHERE __hf_fts_score IS NOT NULL ORDER BY __hf_index_id;"
+    "SELECT * EXCLUDE ({HF_FTS_SCORE}) FROM (SELECT *, fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) AS {HF_FTS_SCORE}"
```
Comment (severo) [inline services/worker/tests/job_runners/split/test_duckdb_index.py:483]:
```suggestion
    f"SELECT * EXCLUDE ({HF_FTS_SCORE}) FROM (SELECT *, fts_main_data.match_bm25({ROW_IDX_COLUMN}, ?) AS {HF_FTS_SCORE}"
```

File: services/worker/tests/job_runners/split/test_duckdb_index.py [hunk 12]
```diff
@@ -479,8 +480,8 @@ def test_get_indexable_columns(features: Features, expected: list[str]) -> None:
...
+    " FROM data) A WHERE {HF_FTS_SCORE} IS NOT NULL ORDER BY {ROW_IDX_COLUMN};"
```
Comment (severo) [inline services/worker/tests/job_runners/split/test_duckdb_index.py:484]:
```suggestion
    f" FROM data) A WHERE {HF_FTS_SCORE} IS NOT NULL ORDER BY {ROW_IDX_COLUMN};"
```

**Review by severo — CHANGES_REQUESTED** [review]
Good PR! I added suggestions (the `f` prefix for f-strings). Also, we automatically order the imports, so: you might have to run `make style` to have it automatically done.

File: services/search/tests/routes/test_filter.py (hunk 3 above)
Comment (severo) [reply services/search/tests/routes/test_filter.py:20]: It's what makes the CI fail:

https://github.com/huggingface/dataset-viewer/actions/runs/9106562973/job/25035748026?pr=2810#step:8:16

```
 src/search/routes/search.py:4:1: I001 [*] Import block is un-sorted or un-formatted
Found 1 error.
[*] 1 fixable with the `--fix` option.
```

File: services/search/tests/routes/test_filter.py [hunk 13]
```diff
@@ -53,10 +53,10 @@ def index_file_location(ds: Dataset) -> Generator[str, None, None]:
     con.execute("LOAD 'fts';")
     con.sql("CREATE OR REPLACE SEQUENCE serial START 0 MINVALUE 0;")
     sample_df = ds.to_pandas()  # noqa: F841
-    create_command_sql = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS __hf_index_id, * FROM sample_df"
+    create_command_sql = f"CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS {ROW_IDX_COLUMN}, * FROM sample_df"
     con.sql(create_command_sql)
     # assert sample_df.shape[0] == con.execute(query="SELECT COUNT(*) FROM data;").fetchall()[0][0]
-    con.sql("PRAGMA create_fts_index('data', '__hf_index_id', 'name', 'gender', overwrite=1);")
+    con.sql(f"PRAGMA create_fts_index('data', ROW_IDX_COLUMN, 'name', 'gender', overwrite=1);")
```
Comment (severo) [inline services/search/tests/routes/test_filter.py:59]:
```suggestion
    con.sql(f"PRAGMA create_fts_index('data', '{ROW_IDX_COLUMN}', 'name', 'gender', overwrite=1);")
```

File: services/search/tests/routes/test_search.py [hunk 14]
```diff
@@ -79,14 +79,14 @@ def test_full_text_search(
         },
         dtype=pd.StringDtype(storage="python"),
     )
-    create_command_sql = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS __hf_index_id, * FROM sample_df"
+    create_command_sql = "CREATE OR REPLACE TABLE data AS SELECT nextval('serial') AS {ROW_IDX_COLUMN}, * FROM sample_df"
     con.sql(create_command_sql)
     con.execute(query="SELECT COUNT(*) FROM data;").fetchall()
     assert sample_df.size == con.execute(query="SELECT COUNT(*) FROM data;").fetchall()[0][0]
-    con.sql("PRAGMA create_fts_index('data', '__hf_index_id', '*', overwrite=1);")
+    con.sql("PRAGMA create_fts_index('data', ROW_IDX_COLUMN, '*', overwrite=1);")
```
Comment (severo) [inline services/search/tests/routes/test_search.py:86]:
```suggestion
    con.sql(f"PRAGMA create_fts_index('data', '{ROW_IDX_COLUMN}', '*', overwrite=1);")
```

File: services/worker/src/worker/job_runners/split/duckdb_index.py [hunk 15]
```diff
@@ -50,10 +51,10 @@
 DATASET_TYPE = "dataset"
 DUCKDB_DEFAULT_INDEX_FILENAME = "index.duckdb"
 DUCKDB_DEFAULT_PARTIAL_INDEX_FILENAME = "partial-index.duckdb"
-CREATE_INDEX_COMMAND = "PRAGMA create_fts_index('data', '__hf_index_id', {columns}, overwrite=1);"
+CREATE_INDEX_COMMAND = f"PRAGMA create_fts_index('data', {ROW_IDX_COLUMN}, {columns}, overwrite=1);"
```
Comment (severo) [inline services/worker/src/worker/job_runners/split/duckdb_index.py:54]:
```suggestion
CREATE_INDEX_COMMAND = f"PRAGMA create_fts_index('data', '{ROW_IDX_COLUMN}', {columns}, overwrite=1);"
```

**Review by severo — CHANGES_REQUESTED** [review]
Some other missing typos

File: services/search/tests/routes/test_filter.py (hunk 3 above)
Comment (devesh-2002) [reply services/search/tests/routes/test_filter.py:20]: > Can you run `make install; make style` in `services/search`, to fix the linting issue (we automatically order the imports)? I can do it for you if you prefer.

Would you please do that?

**Review by severo — DISMISSED** [review]

**Comment by AndreaFrancis** [comment]
It looks good to me; great job @devesh-2002!

**Review by AndreaFrancis — APPROVED** [review]

**Comment by devesh-2002** [comment]
Thanks a lot for all the help! @severo @AndreaFrancis

**Pull request merged**